Makefile

```diff
@@ -11,86 +11,33 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-TEST_ARTIFACTS = prometheus prometheus.race search_index
+GO   := GO15VENDOREXPERIMENT=1 go
+pkgs  = $(shell $(GO) list ./... | grep -v /vendor/)
 
-include Makefile.INCLUDE
+all: format build test
 
-all: binary test
+test:
+	@echo ">> running tests"
+	@$(GO) test -short $(pkgs)
 
-$(GOCC): $(BUILD_PATH)/cache/$(GOPKG)
-	tar -C $(BUILD_PATH)/root -xzf $<
-	touch $@
+format:
+	@echo ">> formatting code"
+	@$(GO) fmt $(pkgs)
 
-advice: $(GOCC)
-	$(GO) vet ./...
+vet:
+	@echo ">> vetting code"
+	@$(GO) vet $(pkgs)
 
-binary: build
-
-build: dependencies $(GOPATH)
-	$(GO) build -o prometheus $(BUILDFLAGS) github.com/prometheus/prometheus/cmd/prometheus
-	$(GO) build -o promtool $(BUILDFLAGS) github.com/prometheus/prometheus/cmd/promtool
+build: 
```

Comment:
I was working on it but had to leave. Will update and look into yours
tomorrow morning.

On Thu, Sep 17, 2015, 9:05 PM Julius Volz <notifications@github.com> wrote:

> In Makefile
> <https://github.com/prometheus/prometheus/pull/1092#discussion_r39785806>:
>
> >
> > -binary: build
> > -
> > -build: dependencies $(GOPATH)
> > -	$(GO) build -o prometheus $(BUILDFLAGS) github.com/prometheus/prometheus/cmd/prometheus
> > -	$(GO) build -o promtool $(BUILDFLAGS) github.com/prometheus/prometheus/cmd/promtool
> > +build:
>
> Trailing whitespace.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/prometheus/prometheus/pull/1092/files#r39785806>.
>